src/languages/cpp.js

```diff
@@ -218,11 +218,31 @@ export default function(hljs) {
     'stderr',
     'stringstream',
     'istringstream',
-    'ostringstream'
+    'ostringstream',
+    'optional',
```

Comment:
Are these keywords or functions?  This is a list of common functions, if this are keywords they don't belong in this list.  It's also possible the list itself needs cleanup if it includes other keywords.

Reply:
It's a mix, so is the already existing part of the list.

E.g. `cos` is a function, `stringstream` is a type etc. Most of the items are functions, but some of them are often-used types. Regarding the interpretation the line is not 100% clear though, because of operator overloading in C++, a static type can be used as though it looks like a function. E.g.:

```c++
std::cout << "Hi there!";
```

Here `std::cout` is a global object of type `ostream` with an overloaded `operator<<()`, making it's use look like a function.

If the list should be functions only, it should definitely be cleaned. I'm willing to pick this up if you'd like me to.

Reply:
Yes, it should be functions... and that list should be used in `FUNCTION_DISPATCH` instead of KEYWORDS (not sure why keywords got there). Any types should be moved out into `TYPES` and we should add them to keywords in addition to the other keys. Any keywords should be moved into KEYWORDS.

```
keywords: { type: TYPES } 
```

CPP_PRIMITIVE_TYPES should probably be `className: 'type'` also.

So we'll probably end up with something like:

```
  const CPP_KEYWORDS = {
    keyword: KEYWORDS,
    built_in: '_Bool _Complex _Imaginary',
    literal: 'true false nullptr NULL ...'
  };

```

Reply:
Hey Josh, I will look into this in the coming days -- I have been having quite a busy life lately ;-)